test/functional/page_objects/visualize_chart_page.ts

```diff
@@ -0,0 +1,399 @@
+      const yAxisRatio = await this.getChartYAxisRatio(axis);
+      const svg = await find.byCssSelector('div.chart');
+      const $ = await svg.parseDomContent();
+      const chartData = $(`g > g.series > rect[data-label="${dataLabel}"]`)
+        .toArray()
+        .map(chart => {
+          const barHeight = Number($(chart).attr('height'));
+          return Math.round(barHeight * yAxisRatio);
+        });
+
+      return chartData;
+    }
+
+    /**
+     * Returns the range/pixel ratio
+     * @param axis axis value, 'ValueAxis-1' by default
+     */
+    private async getChartYAxisRatio(axis = 'ValueAxis-1') {
+      // 1). get the maximum chart Y-Axis marker value and Y position
+      const maxYAxisChartMarker = await retry.try(
+        async () =>
+          await find.byCssSelector(
+            `div.visAxis__splitAxes--y > div > svg > g.${axis} > g:last-of-type.tick`
+          )
+      );
+      const maxYLabel = (await maxYAxisChartMarker.getVisibleText()).replace(/,/g, '');
+      const maxYLabelYPosition = (await maxYAxisChartMarker.getPosition()).y;
+      log.debug(`maxYLabel = ${maxYLabel}, maxYLabelYPosition = ${maxYLabelYPosition}`);
+
+      // 2). get the minimum chart Y-Axis marker value and Y position
+      const minYAxisChartMarker = await find.byCssSelector(
+        'div.visAxis__column--y.visAxis__column--left  > div > div > svg:nth-child(2) > g > g:nth-child(1).tick'
+      );
+      const minYLabel = (await minYAxisChartMarker.getVisibleText()).replace(',', '');
+      const minYLabelYPosition = (await minYAxisChartMarker.getPosition()).y;
+      return (Number(maxYLabel) - Number(minYLabel)) / (minYLabelYPosition - maxYLabelYPosition);
+    }
+
+    public async toggleLegend(show = true) {
+      await retry.try(async () => {
+        const isVisible = find.byCssSelector('.visLegend');
+        if ((show && !isVisible) || (!show && isVisible)) {
+          await testSubjects.click('vislibToggleLegend');
+        }
+      });
+    }
+
+    public async filterLegend(name: string) {
+      await this.toggleLegend();
+      await testSubjects.click(`legend-${name}`);
+      const filters = await testSubjects.find(`legend-${name}-filters`);
+      const [filterIn] = await filters.findAllByCssSelector(`input`);
+      await filterIn.click();
+      await this.waitForVisualizationRenderingStabilized();
+    }
+
+    public async doesLegendColorChoiceExist(color: string) {
+      return await testSubjects.exists(`legendSelectColor-${color}`);
+    }
+
+    public async selectNewLegendColorChoice(color: string) {
+      await testSubjects.click(`legendSelectColor-${color}`);
+    }
+
+    public async doesSelectedLegendColorExist(color: string) {
+      return await testSubjects.exists(`legendSelectedColor-${color}`);
+    }
+
+    public async expectError() {
+      await testSubjects.existOrFail('visLibVisualizeError');
+    }
+
+    public async getChartTypes() {
```

Comment:
I think this should be in visualize_page

Reply:
You are right, fixing it